### PR TITLE
DROOLS-3251 [DMN Designer] Question and Allowed answers not marshalled

### PR DESCRIPTION
**Pull request by tarilabs**
/cc @jomarko @manstis

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 1]
```diff
@@ -460,6 +460,23 @@ public void testEdgewaypoint() throws Exception {
...
+
+    @SuppressWarnings("unchecked")
+    private void checkDecisionqa(Graph<?, Node<?, ?>> graph) {
+        Node<?, ?> decision = graph.getNode("_7052d0f6-ccee-462b-bd89-76afc3b6f67b");
+        assertNodeContentDefinitionIs(decision,
+                                      Decision.class);
+        Decision decisionDefinition = ((View<Decision>) decision.getContent()).getDefinition();
+        assertEquals("hardcoded question", decisionDefinition.getName().getValue());
+        assertEquals("What is the codename?", decisionDefinition.getQuestion().getValue());
+        assertEquals("47.", decisionDefinition.getAllowedAnswers().getValue());
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: Is this a spurious `.` ('47.')? The XML doesn't appear to include it.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java [hunk 2]
```diff
@@ -67,6 +67,8 @@ public DecisionConverter(final FactoryManager factoryManager) {
                                          new BackgroundSet(),
                                          new FontSet(),
                                          new RectangleDimensionsSet());
+        decision.setQuestion(new Question(dmn.getQuestion()));
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java]: Can DMN's value be `null`?

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java [hunk 3]
```diff
@@ -67,6 +67,8 @@ public DecisionConverter(final FactoryManager factoryManager) {
...
+        decision.setAllowedAnswers(new AllowedAnswers(dmn.getAllowedAnswers()));
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java]: Can DMN's value be `null`?

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java [hunk 4]
```diff
@@ -96,6 +98,8 @@ public DecisionConverter(final FactoryManager factoryManager) {
             expression.setParent(d);
         }
         d.setExpression(expression);
+        d.setQuestion(source.getQuestion().getValue());
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java]: If DMN's value can be `null` then we might want to handle an empty `String` (from the UI) as `null`?

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java [hunk 5]
```diff
@@ -96,6 +98,8 @@ public DecisionConverter(final FactoryManager factoryManager) {
...
+        d.setAllowedAnswers(source.getAllowedAnswers().getValue());
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionConverter.java]: If DMN's value can be `null` then we might want to handle an empty `String` (from the UI) as `null`?

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java (hunk 1 above)
Comment (tarilabs) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: image

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: Thanks I was looking at [this](https://github.com/kiegroup/kie-wb-common/blob/a920d00809233600608c7b2781ffffc3fabd1675/kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/resources/decisionqa.dmn#L9). Doh.

Comment (tarilabs) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: np

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by tarilabs** [comment]
```
Test Result (2 failures / +1)
org.kie.workbench.common.screens.server.management.backend.EmbeddedControllerIT.testEmbeddedEndpointsUsingRest
org.kie.workbench.common.screens.server.management.backend.EmbeddedControllerIT.testEmbeddedEndpointsUsingWebSocket
```

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by tarilabs** [comment]
Can this be merged, please.

**Pull request merged**